criu/cr-check.c

```diff
@@ -1665,36 +1673,43 @@ static int pr_set_dumpable(int value)
 
 int check_caps(void)
 {
-	struct proc_status_creds creds;
-	int exit_code = -1;
-
-	if (parse_pid_status(PROC_SELF, &creds.s, NULL))
+	/* Read out effective capabilities and store in opts.cap_eff. */
+	if (set_opts_cap_eff())
 		goto out;
 
-	memcpy(&opts.cap_eff, &creds.cap_eff, sizeof(u32) * PROC_CAP_SIZE);
-
+	/*
+	 * No matter if running as root or not. CRIU always needs
+	 * at least these capabilities.
+	 */
 	if (!has_cap_checkpoint_restore(opts.cap_eff))
 		goto out;
 
 	/* For some things we need to know if we are running as root. */
 	opts.uid = geteuid();
 
-	if (opts.uid) {
```

Comment:
I think we have to check opts.unprivileged instead of opts.uid in all places.  The user can run criu under non-root by mistake and opts.unprivileged guarantees that it is intentional.

Reply:
Done, thanks.

I've left one in place in `check_caps()` so we can check that `opts.unprivileged` is being used in non-root mode.